lang/java/avro/src/main/java/org/apache/avro/reflect/ReflectData.java

```diff
@@ -397,10 +389,12 @@ private FieldAccessor[] getAccessorsFor(Schema schema) {
     }
 
     private FieldAccessor[] createAccessorsFor(Schema schema) {
+
+      var byNameSchema = buildByName(clazz, schema);
```

Comment:
I am not sure how often `#createAccessorsFor(Schema)` is called but `buildByName(clazz, schema)` seems to be wasteful. It uses reflection to extract the field names -> accessor map and then drops it. Next time this method is used again it does it again.

Reply:
Is only called by `#getAccessorsFor(Schema schema` which caches this